ioeffect/src/main/scala/scalaz/ioeffect/RTS.scala

```diff
@@ -42,6 +43,20 @@ trait RTS {
     }
   }
 
+  /**
+   * WARNING: UNSAFE - convert between IO[?, A] and Future - do not use this method.
+   * Seriously, try not to.
+   */
+  final def accursedConversionToFuture[E, A](io: IO[E, A]): concurrent.Future[E \/ A] = {
+    val p = concurrent.Promise[E \/ A] // we must store the information associated with E in \/
+    unsafePerformIOAsync(io.attempt[Throwable])({
+      case ExitResult.Completed(a)  => p.success(a)
+      case ExitResult.Failed(e)     => p.failure(e)
+      case ExitResult.Terminated(t) => p.failure(t)
+    })
+    p.future
+  }
```

Comment:
We return `E \/ A`, so as not to discard the information of the non-terminal error state of the `IO` - there is information in the `E` that we may need, which signals error, not failure. The `E \/ A` is stored in `p.success`, while the rest are piped to `p.failure`